common/src/service/HttpClientHelper.cs

```diff
@@ -701,19 +736,21 @@ static void InsertEtag(HttpRequestMessage requestMessage, string etag)
         {
             cancellationToken.ThrowIfCancellationRequested();
 
-            using (var httpClient = BuildHttpClient(Timeout.InfiniteTimeSpan))
```

Comment:
don't build now, instead use the cached one without timeout, because this one relies on a custom cancellation token